src/auth/authOS.ts

```diff
@@ -0,0 +1,212 @@
+import {useMutation, UseMutationOptions, useQuery, UseQueryOptions} from 'react-query'
+
+import {AUTH_METHOD_KEY, AUTH_METHOD_PIN, ENCRYPTED_PIN_HASH_KEY, useMutationWithInvalidations} from '../hooks'
+import {WalletMeta} from '../legacy/state'
+import storage from '../legacy/storage'
+import {Storage} from '../Storage'
+import {WalletJSON, walletManager, YoroiWallet} from '../yoroi-wallets'
+import {AuthMethod} from '../yoroi-wallets/types'
+import * as Keychain from './Keychain'
+
+export const useCanEnableAuthOs = (options?: UseQueryOptions<boolean, Error>) => {
+  const query = useQuery({
+    queryKey: ['canEnableAuthOs'],
+    queryFn: Keychain.canEnableAuthOs,
+    ...options,
```

Comment:
does this need to be refetched when the app is blurred/focused?

Reply:
yap, it does, if the user disables the auth with OS we need to logout, and ask for a pin creation